locations/spiders/red_roof_us.py

```diff
@@ -0,0 +1,87 @@
+import re
+from datetime import datetime
+
+import chompjs
+from scrapy.spiders import SitemapSpider
+
+from locations.categories import Drink, Extras, apply_yes_no
+from locations.dict_parser import DictParser
+from locations.react_server_components import parse_rsc
+
+BRANDS = {
+    "RRI": {"brand": "Red Roof Inn", "brand_wikidata": "Q7304949"},
+    "RRIPLUS": {"brand": "Red Roof PLUS+"},
+    "REDCOLLECT": {"brand": "Red Collection"},
+    "HTS": {"brand": "HomeTowne Studios", "brand_wikidata": "Q109868848"},
+}
+
+
+def get_time_from_iso(s):
+    if s:
+        return datetime.fromisoformat(s).strftime("%H:%M")
+    else:
+        return None
+
+
+class RedRoofUSSpider(SitemapSpider):
+    name = "red_roof_us"
+    item_attributes = {
+        "brand_wikidata": "Q7304949",
+    }
+    sitemap_urls = ["https://www.redroof.com/sitemap.xml"]
+    sitemap_rules = [
+        (r"^https://www.redroof.com/[\w/]*property/\w{2}/[\w-]+/(\w+)", "parse"),
+    ]
+
+    def parse(self, response):
+        scripts = response.xpath("//script[starts-with(text(), 'self.__next_f.push')]/text()").getall()
+        objs = [chompjs.parse_js_object(s) for s in scripts]
+        rsc = "".join([s for n, s in objs]).encode()
+        data = dict(parse_rsc(rsc))
+
+        pms_data = DictParser.get_nested_key(data, "pmsData")
+        item = DictParser.parse(pms_data)
+        item.update(BRANDS.get(pms_data["brand"], {}))
+        item["extras"]["check_in"] = get_time_from_iso(pms_data["checkInTime"])
+        item["extras"]["check_out"] = get_time_from_iso(pms_data["checkOutTime"])
+        item["name"] = pms_data["description"]
+        item["addr_full"] = pms_data["displayAddressWithZip"]
+        item["extras"]["fax"] = pms_data["faxNumber"]
+        item["ref"] = pms_data["propertyId"]
+        item["street_address"] = pms_data["street1"]
+        if pms_data["propertyName"] != pms_data["description"]:
+            item["branch"] = pms_data["propertyName"]
+
+        item["website"] = response.url
+        if not item["ref"]:
+            for rule in self.sitemap_rules:
+                if match := re.search(rule[0], url):
```

Comment:
`url` is not defined. Did you mean `response.url`?

Reply:
Whoops! I copied this over from `StructuredDataSpider` because I had gotten a few partial responses, but never tested it.